flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/calcite/PreValidateReWriter.scala

```diff
@@ -272,7 +272,7 @@ object PreValidateReWriter {
       targetPosition: util.List[Int]): util.ArrayList[SqlNode] = {
     val targetList = new Array[SqlNode](sourceList.size())
     0 until sourceList.size() foreach {
-      idx => targetList(targetPosition.get(idx)) = sourceList.get(idx)
+      idx => targetList(idx) = sourceList.get(targetPosition.get(idx))
```

Comment:
nit: would be better to add a comment to describe what's the meaning of index and value of the `targetPosition`. I have to deep think it every time looking at this code.

Reply:
Actually, here whole method is `targetPosition.map(sourceList.get)`....